lib/surface/components/dynamic_component.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Surface.Components.Component do
```

Comment:
Should we add a 'props' prop to allow passing a map of props? My thinking is that otherwise users would get warnings about unknown props for this component

Reply:
They shouldn't get any warning but they can also use `{...}`. However, I'm not sure yet if that API is the best one. I'll create some tests for those cases and think how we can improve it.